lib/gui/app/components/safe-webview.js

```diff
@@ -191,12 +191,18 @@ class SafeWebview extends react.PureComponent {
    * @param {Event} event - Event object
    */
   didGetResponseDetails (event) {
-    const HTTP_OK = 200
-    const HTTP_ERR = 400
+    // This seems to pick up all requests related to the webview,
+    // only care about this event if it's a request for the main frame
+    if (event.resourceType === 'mainFrame') {
+      const HTTP_OK = 200
+      const HTTP_ERR = 400
 
-    this.setState({
-      shouldShow: event.httpResponseCode >= HTTP_OK && event.httpResponseCode < HTTP_ERR
-    })
+      analytics.logEvent(event)
+
+      this.setState({
+        shouldShow: event.httpResponseCode >= HTTP_OK && event.httpResponseCode < HTTP_ERR
```

Comment:
We should only show the banner if the HTTP status code is 200, and no other – also, @Shou did you check the Cloudflare docs for the error states when the backing server is unavailable?